src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -103,8 +103,8 @@ def __post_init__(self) -> None:
         metadata["format"] = self.fmt
         metadata["layout"] = self.layout
         metadata["unit"] = self.dataio.unit or ""
-        metadata["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
-        metadata["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
+        metadata["vertical_domain"] = self.dataio.vertical_domain
+        metadata["domain_reference"] = self.dataio.domain_reference
```

Comment:
Much nicer 😄

Reply:
`datum` could possibly also be used. Is it a risk that we will have different (whitelisted) allowed domain_references for each content? E.g. "rkb" would make sense for "content: wellpath", but not for "content: field_outline". Would this be messy? (Perhaps that was the reasoning behind the dict input 🤷‍♂️)

Reply:
> Different topic, but this is also an example of how the data model struggle to scale beyond data objects containing a single dataset. E.g. for a depth surface, this is fine. But for a table containing different things which have different references, it doesn't work. (Same as with "unit".)

I think we have to live with that. I think the only way to mitigate that may be something like domain_reference="multiple" og "n/a" at top level, and then give  spesific data per "sub-type" (like columns in a table).

Reply:
> `datum` could possibly also be used. Is it a risk that we will have different (whitelisted) allowed domain_references for each content? E.g. "rkb" would make sense for "content: wellpath", but not for "content: field_outline". Would this be messy? (Perhaps that was the reasoning behind the dict input 🤷‍♂️)

Would data like "field-outline" actually need a domain reference? Since the data itself is "2D" (bird view)

Reply:
I don't think the dict input would help in the matter of contents having different allowed references.. it was just a way of setting the vertical_domain together with the reference. 
And at the moment I find it difficult to put any logic on the content 🤔 .. some of the contents are quite clear and should be obvious what data it is providing, but many of our contents are general and can be attached to various data... My hope is that much of this gets clearer once we start focusing on 'products' 🤞